advanced_start/workflow_power.rst

```diff
@@ -0,0 +1,157 @@
+
+To create a workflow module, we create a new directory and use the ``init`` command.
+
+.. code :: bash
+  
+    ~/ $ mkdir irisworkflow; cd irisworkflow
+    ~/irisworkflow/ $ nstack init workflow 
+    Workflow module 'irisworkflow' successfully initialised at /var/home/fedora/irisworkflow
+
+NStack creates a single file, ``workflow.nml``, which contains a sample workflow. 
+
+.. code :: java
+
+  module irisworkflow {                                                                                                                           
+    def w = sources.http : Text { http_path = "/foo" } | Module1.numChars | sinks.log : Integer;
+  }                                                                             
+
+In this sample, we have a module called ``irisworkflow`` which has a single workflow on it, ``w``. If we replace ``Module.numChars`` with the function from our previous tutorial, ``demo.numChars``, we can then build this workflow with ``nstack build``.
+
+
+.. code :: bash
+ 
+  ~/irisworkflow/ $ nstack build
+  Building NStack Workflow module irisworkflow.
+  Workflow module irisworkflow built successfully.
+
+And now start this directly with the start command:
+
+.. code :: bash
+
+  ~/irisworkflow/ $ nstack start irisworkflow.w
+
+Multiple steps
+**************
+
+Workflows can contain as many steps as you like, as long as the output type of one matches the input type of the other. For instance, let's say we wanted to create the following workflow:
+
+- Expose an HTTP endpoint which takes four ``Double``\s
+- Send these ``Double``\s to our classifier, ``irisclassify``, which will tell us the species of the iris
+- Count the number of characters in the species of the iris
+- Write the result to the log
+
+We could write the following workflow:
+
+.. code :: bash
+   
+  module irsiworkflow {
+    def multipleSteps = sources.http (Double, Double, Double, Double) { http_path = "/irisendpoint" } | irisclassify.predict | demo.numChars | sinks.log : Integer;
+  }
+
+.. note :: ``numChars`` and ``predict`` can be `composed` together because their types -- or schemas -- match. If ``predict`` wasn't configured to output ``Text``, or ``numChars`` wasn't configured to take ``Text`` as input, NStack would not let you build the following workflow.
+
+Partial workflows
+*****************
+
+All of the workflows that we have written so far have been `fully composed`, which means that they contain a source, one or more functions, and a sink. Many times, you want to split up sources, sinks, and functions into separate pieces you can share and reuse. In this case, we say that a workflow is `partially composed`, which just means it does not contain a source, one or more functions, and a sink. These workflows cannot be ``start``\ed by themselves, but can be shared and attached to other sources, sinks, or functions to become `fully composed`. 
```

Comment:
OK, so "one or more functions" could be "zero or more functions"? (but would suggest tweaking the wording)